frontend/src/components/FileUpload.tsx

```diff
@@ -64,4 +65,56 @@ const FileUpLoad = (props: UploadProps) => {
   );
 };
 
+// FileUploadHOF is a higher order function. It accepts a CB and returns
+// the FileUpload component. The difference from the above is that it will
+// run whatever you specified in the CB after a file is successfully uploaded
+// and the file URL is received.
+export const FileUploadHOF = (cb?: (url: string, type?: string) => void) => {
```

Comment:
Very nice work. 
Since this isn't really a HOF but a custom hook the naming convention should be useFileUpload or something along those lines as per React documentation.